com.dynamo.cr/com.dynamo.cr.bob/src/com/dynamo/bob/util/BuildInputDataCollector.java

```diff
@@ -0,0 +1,137 @@
+package com.dynamo.bob.util;
+
+import com.dynamo.bob.archive.EngineVersion;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.FileWriter;
+import java.nio.file.Files;
+import java.security.MessageDigest;
+
+public class BuildInputDataCollector {
+    private static final String DATA_FILE_NAME = "build_input_data.json";
+
+    private static String[] bobArguments;
+    private static Map<String, File> dependencies;
+
+    public static void setArgs(String[] args) {
+        bobArguments = args;
+    }
+
+    public static void setDependencies(Map<String, File> libFiles) {
+        dependencies = libFiles;
+    }
+
+    public static void saveDataAsJson(String rootDirectory, File bundleOutputDirectory) {
+        TimeProfiler.start("BuildInputDataCollector");
+        Map<String, Object> data = new HashMap<>();
+        data.put("BobArguments", bobArguments);
+
+        Map<String, String> engineVersion = new HashMap<>();
+        engineVersion.put("sha1", EngineVersion.sha1);
+        engineVersion.put("version", EngineVersion.version);
+        engineVersion.put("timestamp", EngineVersion.timestamp);
+        data.put("EngineVersion", engineVersion);
+        data.put("Environment", System.getenv());
+
+        Map<String, String> gitInfo = getGitInfoIfAvailable(rootDirectory);
+        if (gitInfo != null) {
+            data.put("ProjectGit", gitInfo);
+        }
+
+        if (dependencies != null && !dependencies.isEmpty()) {
+            List<Map<String, String>> depsList = new ArrayList<>();
+            for (Map.Entry<String, File> entry : dependencies.entrySet()) {
+                File file = entry.getValue();
+                if (file.exists() && file.isFile()) {
+                    try {
+                        byte[] fileBytes = Files.readAllBytes(file.toPath());
+                        MessageDigest md = MessageDigest.getInstance("MD5");
+                        byte[] digest = md.digest(fileBytes);
+                        StringBuilder hexString = new StringBuilder();
+                        for (byte b : digest) {
+                            String hex = Integer.toHexString(0xff & b);
+                            if (hex.length() == 1) hexString.append('0');
+                            hexString.append(hex);
+                        }
+                        Map<String, String> depEntry = new HashMap<>();
```

Comment:
Don't we already have a helper function for this? 🤔

Reply:
true!